packages/container/lib/container.js

```diff
@@ -175,10 +255,31 @@ Container.prototype = {
   }
 };
 
+if (isFeatureEnabled('ember-factory-for')) {
+  /**
+   Given a fullName, return the corresponding factory. The consumer of factory is repsonsible
+   for the destruction of the factory as there is no way to understand the objects lifecycle.
```

Comment:
"for the destruction of the factory" -> "for the destruction of any factory instances, as there is no way for the container to ensure instances are destroyed when it itself is destroyed"